src/read.c

```diff
@@ -2663,13 +2641,8 @@ UInt ReadEvalFile(Obj *evalResult)
     IntrFuncExprBegin(0, nloc, nams, GetInputLineNumber());
 
     /* read the statements                                                 */
-    {
-        UInt oldLoopNesting = rs->LoopNesting;
-        rs->LoopNesting = 0;
-        nr = ReadStats(s, S_SEMICOLON | S_EOF);
-        rs->LoopNesting = oldLoopNesting;
-    }
-
+    GAP_ASSERT(rs->LoopNesting == 0);
+    nr = ReadStats(rs, S_SEMICOLON | S_EOF);
```

Comment:
Real change: since we freshly allocated the `ReaderState` on the stack, there is no need to "restore" its `LoopNesting` member here.